File: src/main/java/responses/user/Image.java
package responses.user;

import com.google.gson.annotations.SerializedName;
import responses.SixtyBySixty;

public class Image {
    @SerializedName("60x60")
    private final SixtyBySixty sixtyBySixty;

    public Image(SixtyBySixty sixtyBySixty) {
        this.sixtyBySixty = sixtyBySixty;
    }

    public SixtyBySixty getSixtyBySixty() {
        return sixtyBySixty;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Image image = (Image) o;

        return !(sixtyBySixty != null ? !sixtyBySixty.equals(image.sixtyBySixty) : image.sixtyBySixty != null);

    }

    @Override
    public int hashCode() {
        return sixtyBySixty != null ? sixtyBySixty.hashCode() : 0;
    }

    @Override
    public String toString() {
        return "Image{" +
                "sixtyBySixty=" + sixtyBySixty +
                '}';
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/responses/pin/Image.kt
package com.chrisdempewolf.pinterest.responses.pin

class Image(val original: Original?) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val image = o as Image?

        return !if (original != null) original != image!!.original else image!!.original != null

    }

    override fun hashCode(): Int{
        return original?.hashCode() ?: 0
    }

    override fun toString(): String {
        return "Image{" +
                "original=" + original +
                '}'
    }
}
